BuildingBlocks/src/BuildingBlocks.API/Mvc/ExceptionFilters/CustomExceptionFilter.cs

```diff
@@ -218,3 +216,72 @@ private IEnumerable<string> GetFormattedStackTrace(Exception exception)
             Regex.Matches(exception.StackTrace, "at .+").Select(m => m.Value.Trim());
     }
 }
+
+file static class LoggerExtensions
+{
+    private static readonly Action<ILogger, InfrastructureException, string, string, Exception> INFRASTRUCTURE_EXCEPTION =
+        LoggerMessage.Define<InfrastructureException, string, string>(
+            LogLevel.Information,
+            new EventId(560507, "ExceptionFilter.InfrastructureException"),
+            "An '{exception}' occurred. Error Code: '{code}'. Error message: '{message}'."
+        );
+
+    private static readonly Action<ILogger, ApplicationException, string, string, Exception> APPLICATION_EXCEPTION =
+        LoggerMessage.Define<ApplicationException, string, string>(
+            LogLevel.Information,
+            new EventId(437832, "ExceptionFilter.ApplicationException"),
+            "An '{exception}' occurred. Error Code: '{code}'. Error message: '{message}'."
+        );
+
+    private static readonly Action<ILogger, DomainException, string, string, Exception> DOMAIN_EXCEPTION =
+        LoggerMessage.Define<DomainException, string, string>(
+            LogLevel.Information,
+            new EventId(505278, "ExceptionFilter.DomainException"),
+            "An '{exception}' occurred. Error Code: '{code}'. Error message: '{message}'."
+        );
```

Comment:
You can combine these three into one (the name can be something like "InvalidUserInput)"